test/e2e/nstemplatetier_test.go

```diff
@@ -87,47 +90,76 @@ func TestNSTemplateTiers(t *testing.T) {
+	// first, let's create the `cheesecake` NSTemplateTier (to avoid messing with other tiers)
+	// We'll use the `basic` tier as a source of inspiration.
+	_, err := hostAwaitility.WaitForNSTemplateTier("basic",
+		UntilNSTemplateTierSpec(Not(HasNamespaceTemplateRefs("basic-code-000000a"))),
+		UntilNSTemplateTierSpec(Not(HasNamespaceTemplateRefs("basic-dev-000000a"))),
+		UntilNSTemplateTierSpec(Not(HasNamespaceTemplateRefs("basic-stage-000000a"))),
+		UntilNSTemplateTierSpec(Not(HasClusterResourcesTemplateRef("basic-clusterresources-000000a"))),
+	)
+	require.NoError(t, err)
+	basic := &toolchainv1alpha1.NSTemplateTier{}
+	err = hostAwaitility.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: hostAwaitility.Ns,
+		Name:      "basic",
+	}, basic)
+	// now let's create the new NSTemplateTier with the same templates as the "basic" tier
+	cheesecake := &toolchainv1alpha1.NSTemplateTier{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: basic.Namespace,
+			Name:      "cheesecake",
+		},
+		Spec: basic.Spec,
+	}
+	err = hostAwaitility.Client.Create(context.TODO(), cheesecake, testsupport.CleanupOptions(ctx))
+	require.NoError(t, err)
+	// now, let's create a few users (more than `maxPoolSize`)
+	count := MaxPoolSize + 1
+	users := make([]toolchainv1alpha1.UserSignup, count)
+	nameFmt := "cheesecakelover%d"
+	for i := 0; i < count; i++ {
+		users[i] = createAndApproveSignup(t, awaitility, fmt.Sprintf(nameFmt, i))
+	}
+	// and wait until there are all provisioned
+	for i := range users {
+		_, err := hostAwaitility.WaitForMasterUserRecord(fmt.Sprintf(nameFmt, i))
+		require.NoError(t, err)
+	}
+	// now, let's promote to users the `cheesecake` tier
+	for i := range users {
+		changeTierRequest := newChangeTierRequest(hostAwaitility.Ns, cheesecake.Name, fmt.Sprintf(nameFmt, i))
+		err = awaitility.Client.Create(context.TODO(), changeTierRequest, &test.CleanupOptions{})
+		require.NoError(t, err)
+		_, err = hostAwaitility.WaitForChangeTierRequest(changeTierRequest.Name, toBeComplete)
+		require.NoError(t, err)
+		_, err := hostAwaitility.WaitForMasterUserRecord(fmt.Sprintf(nameFmt, i))
+		require.NoError(t, err)
+	}
 
-		// then
-		verifyMurIsProvisionedForRevisionMadeOutOfOtherTier(t, awaitility, murName, "basic", "789ijkl")
-	})
+	// now, let's update the NSTemplateTier using the namespace templates of the `advanced` tier
+	advanced := &toolchainv1alpha1.NSTemplateTier{}
+	err = hostAwaitility.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: hostAwaitility.Ns,
+		Name:      "advanced",
+	}, advanced)
+	require.NoError(t, err)
+	// when
+	cheesecake.Spec.ClusterResources = advanced.Spec.ClusterResources
+	cheesecake.Spec.Namespaces = advanced.Spec.Namespaces
+	err = hostAwaitility.Client.Update(context.TODO(), cheesecake)
+	// then
+	require.NoError(t, err)
+	// and there should be `MaxPoolSize` TemplateUpdateRequests created
+	err = hostAwaitility.WaitForTemplateUpdateRequests(hostAwaitility.Ns, MaxPoolSize)
```

Comment:
Also let's check that the corresponding NamespaceTemplateSet was actually updated and now both Cluster and Namespace templates are using the updated templates.

Also, I would add some "noise". Users/Murs with tiers we are not updating. It will test all our label/selector dance.

Reply:
> Also let's check that the corresponding NamespaceTemplateSet was actually updated and now both Cluster and Namespace templates are using the updated templates.

where, in the MasterUserRecords? That's not implemented yet. This PR is for https://issues.redhat.com/browse/CRT-683 (which deals with creating all the `TemplateUpdateRequest` resources), whereas the update of the `MasterUserRecord` resources will be covered in https://issues.redhat.com/browse/CRT-685

Reply:
Ah, right.